integration_tests/suite/helpers/fixtures.py

```diff
@@ -1,29 +1,46 @@
-# Copyright 2017 The Wazo Authors  (see the AUTHORS file)
+# Copyright 2017-2019 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+import copy
+import time
+import requests
 from functools import wraps
 from wazo_webhookd_client.exceptions import WebhookdError
+from xivo_test_helpers import until
 
 from .base import VALID_TOKEN
 
 
-def subscription(subscription_args):
+def subscription(subscription_args, track_test_name=True):
     '''This decorator is only compatible with instance methods, not pure functions.'''
     def decorator(decorated):
         @wraps(decorated)
         def wrapper(self, *args, **kwargs):
             webhookd = self.make_webhookd(VALID_TOKEN)
-            new_subscription = webhookd.subscriptions.create(subscription_args)
-            args = list(args) + [new_subscription]
-            try:
-                result = decorated(self, *args, **kwargs)
-            finally:
+
+            sub = copy.deepcopy(subscription_args)
+            if track_test_name:
+                # Add test name to help debugging
+                sep = "&" if "?" in sub['config']['url'] else "?"
+                sub['config']['url'] += sep + "test_case=" + decorated.__name__
+
+            new_subscription = webhookd.subscriptions.create(sub)
+            self.ensure_webhookd_consume_uuid(new_subscription['uuid'])
+
+            def cleanup():
                 try:
                     webhookd = self.make_webhookd(VALID_TOKEN)
                     webhookd.subscriptions.delete(new_subscription['uuid'])
                 except WebhookdError as e:
                     if e.status_code != 404:
                         raise
-            return result
+
+                self.ensure_webhookd_not_consume_uuid(new_subscription['uuid'])
+
+            self.addCleanup(cleanup)
+
+            args = list(args) + [new_subscription]
```

Comment:
I like that you added the new subscriptions at the end of the args, it feels more natural when using the decorator.
I think we should adopt this style for most of our test fixtures. 

We usually do `decorated(self, new_suscription, *args, **kwargs)`

Reply:
The diff is confusing, but that's one is not mine :)